tests/tf/algos/test_trpo.py

```diff
@@ -17,29 +19,30 @@
 class TestTRPO(unittest.TestCase):
     def test_trpo_pendulum(self):
         """Test TRPO with Pendulum environment."""
-        env = TfEnv(normalize(gym.make("Pendulum-v0")))
-        policy = GaussianMLPPolicy(
-            name="policy",
-            env_spec=env.spec,
-            hidden_sizes=(32, 32),
-            hidden_nonlinearity=tf.nn.tanh,
-            output_nonlinearity=None,
-        )
-        baseline = GaussianMLPBaseline(
-            env_spec=env.spec,
-            regressor_args=dict(hidden_sizes=(32, 32)),
-        )
-        algo = TRPO(
-            env=env,
-            policy=policy,
-            baseline=baseline,
-            batch_size=1024,
-            max_path_length=100,
-            n_itr=10,
-            discount=0.99,
-            gae_lambda=0.98,
-            policy_ent_coeff=0.0,
-            plot=False,
-        )
-        last_avg_ret = algo.train()
-        assert last_avg_ret > -400
+        with tf.Graph().as_default():
```

Comment:
perhaps you can put this into `self.setUp()` and or `self.tearDown()`, which are run before/after each test case method respectively. The purpose of setUp/tearDown is exactly this case. It might be cleaner to make a new tf.Session() for each test case, for instance?

https://docs.python.org/3.5/library/unittest.html#unittest.TestCase.setUp